uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/apis/work-pattern-api_test.js

```diff
@@ -1,53 +1,83 @@
+/** eslint-env amd, jasmine */
+
 define([
+  'common/lodash',
   'mocks/data/work-pattern-data',
   'leave-absences/shared/apis/work-pattern-api',
-], function (mockData) {
+], function (_, mockData) {
   'use strict';
 
   describe('WorkPatternAPI', function () {
     var WorkPatternAPI, $httpBackend;
 
     beforeEach(module('leave-absences.apis'));
-
     beforeEach(inject(function (_WorkPatternAPI_, _$httpBackend_) {
       WorkPatternAPI = _WorkPatternAPI_;
       $httpBackend = _$httpBackend_;
-
-      //Intercept backend calls for WorkPatternAPI.getCalendar
-      $httpBackend.whenGET(/action\=getcalendar&entity\=WorkPattern/)
-        .respond(mockData.daysData());
     }));
 
-    describe('calendarData()', function () {
-      var workPatternPromise,
-        dummyContactId,
-        dummyPeriodId;
+    afterEach(function () {
+      $httpBackend.flush();
+    });
+
+    describe('getCalendar()', function () {
+      var workPatternPromise;
+      var dummyContactId = 1;
+      var dummyPeriodId = 2;
+      var additionalFilters = { foo: 'foo', bar: 'bar' }
 
       beforeEach(function () {
+        $httpBackend.whenGET(/action\=getcalendar&entity\=WorkPattern/).respond(mockData.daysData());
         spyOn(WorkPatternAPI, 'sendGET').and.callThrough();
-        dummyContactId = "1";
-        dummyPeriodId = "2";
-        workPatternPromise = WorkPatternAPI.getCalendar(dummyContactId, dummyPeriodId, jasmine.any(Object));
       });
 
-      afterEach(function () {
-        $httpBackend.flush();
-      });
+      describe('basic tests', function () {
+        beforeEach(function () {
+          workPatternPromise = WorkPatternAPI.getCalendar(dummyContactId, dummyPeriodId, additionalFilters);
+        });
 
-      it('calls the sendGET() method', function () {
-        expect(WorkPatternAPI.sendGET).toHaveBeenCalled();
-        expect(WorkPatternAPI.sendGET.calls.mostRecent().args[0]).toBe('WorkPattern');
-        expect(WorkPatternAPI.sendGET.calls.mostRecent().args[1]).toBe('getcalendar');
-      });
+        it('sends a GET request to the api', function () {
+          expect(WorkPatternAPI.sendGET).toHaveBeenCalled();
```

Comment:
These are two distinct tests
1. Tests if a given GET request is called
2. Tests if the GET request has been done with specific parameters

The second test in fact takes for granted that the GET request happened `WorkPatternAPI.sendGET.calls.mostRecent()`.

I thought of leaving it like this:
```js
it(/* .. */, function () {
  expect(WorkPatternAPI.sendGET).toHaveBeenCalled();
  expect(WorkPatternAPI.sendGET.calls.mostRecent().args[0]).toBe('WorkPattern');
  expect(WorkPatternAPI.sendGET.calls.mostRecent().args[1]).toBe('getcalendar');
});
```
but at that point I preferred having the test suite being more expressive, hence the two tests